hs/CalendarResponse.hs

```diff
@@ -0,0 +1,281 @@
+
+-- | Obtains a list with all the information about the courses obtained from 
+-- the cookies.
+getCoursesInfo :: String -> [(String, String, String)]
+getCoursesInfo courses = map courseInfo allCourses
+    where
+        courseInfo course = (course !! 0, course !! 1, course !! 2)
+        allCourses = map (splitOn "-") byCourse
+        byCourse = splitOn "_" courses
+
+-- ** Database Information
+
+-- | Pulls out the information for each course from the database.
+pullDatabase :: (String, String, String) -> IO ([Time], String, String, String)
+pullDatabase (code, sect, session) =
+    if take 1 sect == "L"
+    then getData $ returnLectureTimes (T.pack code)
+                                      (T.pack sect)
+                                      (T.pack session)
+    else getData $ returnTutorialTimes (T.pack code)
+                                       (T.pack sect)
+                                       (T.pack session)
+    where
+        getData info = do
+            courseInfo <- info 
+            return (courseInfo, code, session, sect)
+
+-- ** Event Creation
+
+-- | Generates an event for each course.
+getEvent :: String -> ([Time], String, String, String) -> [String]
+getEvent timeSystem (timeFields, code, session, sect) = 
+    concat $ eventsByCourse code session sect start end dates timeSystem
+    where
+        dataInOrder = orderData timeFields
+        start = startTime dataInOrder
+        end = endTime dataInOrder
+        dates = getDatesByCourse session dataInOrder
+
+-- | Creates an event for each course depending on its session.
+eventsByCourse :: String -- ^ Course code.
+            -> String -- ^ Course session.
+            -> String -- ^ Course section.
+            -> [[String]] -- ^ Start times.
+            -> [[String]] -- ^ End times.
+            -> [[(String, String)]] -- ^ Start/End dates.
+            -> String -- ^ Current time.
+            -> [[String]]
+eventsByCourse code session sect start end dates timeSystem =
+    if session == "Y" then year else halfCourse
+    where
+        year = halfFall ++ halfWinter
+        halfFall = eventsByTime code sect start end (dates !! 0) timeSystem
+        halfWinter = eventsByTime code sect start end (dates !! 1) timeSystem
+        halfCourse = eventsByTime code sect start end (dates !! 0) timeSystem
+
+-- | Creates an event for each start/end time and date.
+eventsByTime :: String -- ^ Course code.
+             -> String  -- ^ Course section.
+             -> [[String]] -- ^ Start times.
+             -> [[String]] -- ^ End times.
+             -> [(String, String)] -- ^ Start/End dates.
+             -> String -- ^ Current time.
+             -> [[String]]
+eventsByTime code sect start end dates timeSystem =
+    map (eventsByDate code sect timeSystem) (zip' start end dates)
+
+-- | Join three lists together in tuples of three elements
+zip' :: [a] -> [b] -> [c] -> [(a,b,c)]
+zip' (x:xs) (y2:ys) (z:zs) = (x,y2,z):zip' xs ys zs
```

Comment:
The compiler was giving me an error because of an existing binding. However, I found a way around by importing only the functions I actually use in my program. Therefore I was able to change y2 for y1.